### ISSUE TITLE
Feature request Cheat Search

### ISSUE THREAD

**Issue by Cibomatto2002** [stella-emu/stella#163]
Most emulators has some kind of Cheat Search will we see one one day? Also any plans to add the Atari 5200 and 7800?

**Comment by sa666666** [comment]
I will look into the cheat stuff when I have more time, but definitely not before the 5.0 release.

As for adding 5200 and 7800 support, probably not ever.  There are other projects that do these, and spreading my time among 3 emulators will result in none of them being good.

**Comment by thrust26** [comment]
An idea:

1. user defines maximum number of value to cheat -> number of bits to search for
2. user sets the current value
3. user plays the game so that the number changes
4. user sets the new, current value
5. Stella presents a list of RAM values which have changed accordingly (using matching deltas and maybe the expected values (+/-1). 
6. if the list is too large the user can decide to repeat from step 3
7. user accepts the list, defines the requested value and the cheats are generated (see below) and added to the game's cheat list (all disabled)
8. user tests the cheats one by one and (hopefully) identifies the correct one
9. user deletes the wrong cheats
10. done! 😄 
* We need a new cheat method. Currently we are patching the ROM. This is way too complicated for the average user and also very complex when we try to do do this automatically. Therefore this approach requires patching the RAM and freezing the relevant bits.

**Comment by sa666666** [comment]
There is already a RAM continuous write cheat.  See `RamCheat.cxx`.  I guess this class can be extended or built upon.

**Comment by thrust26** [comment]
Yes, that would work well. We would have to extend it to support extended RAM and RAM banks.
